core/core.go

```diff
@@ -0,0 +1,90 @@
+package core
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"reflect"
+	"time"
+
+	"github.com/mitre/gocat/agent"
+	"github.com/mitre/gocat/output"
+
+	_ "github.com/mitre/gocat/execute/shells" // necessary to initialize all submodules
+)
+
+// Initializes and returns sandcat agent.
+func initializeCore(server string, group string,c2 map[string]string, p2pReceiversOn bool, verbose bool) (*agent.Agent, error) {
+	output.SetVerbose(verbose)
+	output.VerbosePrint("Starting sandcat in verbose mode.")
+	return agent.AgentFactory(server, group, c2, p2pReceiversOn)
+}
+
+//Core is the main function as wrapped by sandcat.go
+func Core(server string, group string, delay int, c2 map[string]string, p2pReceiversOn bool, verbose bool) {
+	http.DefaultTransport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
+	sandcatAgent, err := initializeCore(server, group, c2, p2pReceiversOn, verbose)
+	if err != nil {
+		output.VerbosePrint(fmt.Sprintf("[-] Error when initializing agent: %s", err.Error()))
+		output.VerbosePrint("[-] Exiting.")
+	} else {
+		sandcatAgent.Display()
+		output.VerbosePrint(fmt.Sprintf("initial delay=%d", delay))
+		time.Sleep(time.Duration(float64(delay)) * time.Second)
```

Comment:
`Agent.Sleep(time)`?

Reply:
I'm adding the delay parameter into the Agent, so I suppose Agent.Sleep would make sense. Or at least Agent.InitialDelay()